backend/src/app.ts

```diff
@@ -0,0 +1,40 @@
+import { swaggerUI } from "@hono/swagger-ui"
+import { OpenAPIHono, z } from "@hono/zod-openapi"
+
+import * as schemas from "./api/components/schemas"
+import problemsApp from "./api/paths/problems"
+import submissionsApp from "./api/paths/submissions"
+
+const app = new OpenAPIHono()
+
+app.route("/api", problemsApp)
+app.route("/api", submissionsApp)
+
+const openApiDocument = {
+  info: {
+    license: { name: "" },
+    title: "Problem and Submission API",
+    version: "0.1.0",
+  },
+  openapi: "3.1.0",
+  paths: {},
+}
+
+app.doc("/api/openapi.json", () => {
+  const generatedSchemas = Object.fromEntries(
+    Object.entries(schemas)
+      .filter(([, schema]) => schema instanceof z.ZodType)
+      .map(([key, schema]) => [key, (schema as z.ZodType).openapi({})]),
+  )
+
+  return {
+    ...openApiDocument,
+    components: {
+      schemas: generatedSchemas,
+    },
+  }
+})
```

Comment:
imo: `schemas`は`@hono/zod-openapi`が自動的に生成するので、手動で書かなくて良いと思われます！

参考文献: https://github.com/honojs/middleware/tree/main/packages/zod-openapi#basic-usage

```suggestion
app.doc("/api/openapi.json", openApiDocument)
```